test/FunctionalTests/Microsoft.OData.Client.Tests/ALinq/SequenceMethodsTests.cs

```diff
@@ -0,0 +1,290 @@
+
+        public SequenceMethodsTests()
+        {
+            _ctx = new TestContext();
+        }
+
+        [Fact]
+        public void Any()
+        {
+            _ctx.InterceptRequestAndAssertUri("/Customers/$count");
+            _ctx.InterceptRequestAndMockResponse("91");
+            Assert.True(_ctx.Customers.Any());
+
+            _ctx.InterceptRequestAndAssertUri("/Customers/$count?$filter=contains(ContactName,'thisdoesntexist')");
+            _ctx.InterceptRequestAndMockResponse("0");
+            Assert.False(_ctx.Customers.Where(c => c.Name.Contains("thisdoesntexist")).Any());
+        }
+
+        [Fact]
+        public void AnyPredicate()
+        {
+            _ctx.InterceptRequestAndAssertUri("/Customers/$count?$filter=contains(ContactName,'ab')");
+            _ctx.InterceptRequestAndMockResponse("6");
+            Assert.True(_ctx.Customers.Any(c => c.Name.Contains("ab")));
+
+            _ctx.InterceptRequestAndAssertUri("/Customers/$count?$filter=contains(ContactName,'thisdoesntexist')");
+            _ctx.InterceptRequestAndMockResponse("0");
+            Assert.False(_ctx.Customers.Any(c => c.Name.Contains("thisdoesntexist")));
+        }
+
+        [Fact]
+        public void CountPredicate()
+        {
+            _ctx.InterceptRequestAndAssertUri("/Customers/$count?$filter=contains(ContactName,'ab')");
+            _ctx.InterceptRequestAndMockResponse("6");
+            int count = _ctx.Customers.Count(c => c.Name.Contains("ab"));
+            Assert.Equal(6, count);
+        }
+
+        [Fact]
+        public void LongCountPredicate()
+        {
+            _ctx.InterceptRequestAndAssertUri("/Customers/$count?$filter=contains(ContactName,'ab')");
+            _ctx.InterceptRequestAndMockResponse("6");
+            long count = _ctx.Customers.LongCount(c => c.Name.Contains("ab"));
+            Assert.Equal(6, count);
+        }
+
+        [Fact]
+        public void FirstPredicate()
+        {
+            // Test behaviour with one match:
+            _ctx.InterceptRequestAndAssertUri("/Customers?$filter=ContactName ne 'John'&$top=1");
+            _ctx.InterceptRequestAndMockResponseValue("Customers", "[" + TestContext.MockCustomer1 + "]");
+            Customer customer = _ctx.Customers.First(c => c.Name != "John");
+            Assert.Equal("ALFKI", customer.Id);
+            Assert.Equal("Maria Anders", customer.Name);
+            Assert.Equal("Berlin", customer.City);
+
+            // Test behaviour with no match:
+            bool expectedExceptionThrown = false;
+            try
+            {
+                _ctx.InterceptRequestAndAssertUri("/Customers?$filter=ContactName eq 'thisdoesntexist'&$top=1");
+                _ctx.InterceptRequestAndMockResponseValue("Customers", "[]");
+                _ctx.Customers.First(c => c.Name == "thisdoesntexist");
+            }
+            catch (InvalidOperationException)
+            {
+                expectedExceptionThrown = true;
+            }
```

Comment:
This could be a separate test method. And we could use `Assert.Throws<InvalidOperationException>` to ensure 

```c#
[Fact]
public void FirstPredicate_ThrowsException_WhenNoMatchExists()
{
     _ctx.InterceptRequestAndAssertUri("/Customers?$filter=ContactName eq 'thisdoesntexist'&$top=1");
      _ctx.InterceptRequestAndMockResponseValue("Customers", "[]");
      Assert.Throws<InvalidOperationException>(()  => _ctx.Customers.First(c => c.Name == "thisdoesntexist"));
}
```